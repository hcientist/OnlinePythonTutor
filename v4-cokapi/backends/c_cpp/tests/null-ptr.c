int main() {
  int* x = malloc(8);
  free(x); // show x pointing to garbage
  x = 0;   // show x pointing to NULL
}
