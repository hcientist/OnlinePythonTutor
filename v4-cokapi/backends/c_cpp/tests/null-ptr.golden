{
  "code": "int main() {\n  int* x = malloc(8);\n  free(x); // show x pointing to garbage\n  x = 0;   // show x pointing to NULL\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000100", 
              "pointer", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 1, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000100", 
              "pointer", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 1, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x53FE040": [
          "C_ARRAY", 
          "0x53FE040", 
          [
            "C_DATA", 
            "0x53FE040", 
            "int", 
            "<UNINITIALIZED>"
          ], 
          [
            "C_DATA", 
            "0x53FE044", 
            "int", 
            "<UNINITIALIZED>"
          ]
        ]
      }, 
      "line": 3, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x53FE040"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 3, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x53FE040": [
          "C_ARRAY", 
          "0x53FE040"
        ]
      }, 
      "line": 4, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x53FE040"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 5, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x0"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

